src/renderer/base/renderer.cpp

```diff
@@ -755,14 +760,59 @@ void Renderer::_PaintBufferOutputHelper(_In_ IRenderEngine* const pEngine,
                 // Otherwise if it's not a special case, just insert it as is.
                 else
                 {
-                    clusters.emplace_back(it->Chars(), it->Columns());
+                    TextBufferCellIterator nextCell = it + static_cast<ptrdiff_t>(std::max<size_t>(1, currentCharCols));
+                    bool expanded = false;
+                    bool inJoiner = nextCell->UnicodeAttr().IsJoiner();
+                    bool moveForward = nextCell && (nextCell->UnicodeAttr().IsBackwordAdhesive());
+
+                    std::wstring* currentText = new std::wstring(it->Chars());

```

Comment:
This is just purely wrong. I couldn't find a way to concat two `std::string_view`. Neither can I access the underlying text buffer. I had to allocate a new string to get the content I need. I'd love to hear some opinions.

Reply:
Perhaps use of the newly added https://github.com/fmtlib/fmt might help?